src/cargo/ops/cargo_package.rs

```diff
@@ -405,6 +429,23 @@ fn warn_on_nonexistent_file(
     ))
 }
 
+fn warn_custom_build_file_not_in_package(
+    pkg: &Package,
+    path: &Path,
+    target: &Target,
+    ws: &Workspace<'_>,
+) -> CargoResult<()> {
+    let msg = format!(
```

Comment:
I would lean toward a hard error. Without build script being packaged this package cannnot even compile.

Reply:
Ok. I will used `shell().error()` instead.

Reply:
By hard error I meant Cargo should stop building with an exit code other than 0. I don't think `shell().error()` doing so. Returning an `Result::Err` from `anyhow` should do that automatically.

Reply:
Oh, i misunderstood. :joy: